core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -120,4 +122,23 @@ public QB readFrom(StreamInput in) throws IOException {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
     }
+
+    protected static List<QueryBuilder> readQueryBuilderListFrom(StreamInput in) throws IOException {
+        List<QueryBuilder> list = new ArrayList<>();
+        int size = in.readInt();
+        for (int i = 0; i < size; i++) {
+            QueryBuilder newBuilder = in.readNamedWriteable();
+            if (newBuilder != null) {
+                list.add(newBuilder);
+            }
+        }
+        return list;
+    }
+
+    protected static void writeQueryBuilderListTo(StreamOutput out, List<QueryBuilder> list) throws IOException {
+        out.writeInt(list.size());
+        for (QueryBuilder queryBuilder : list) {
+            out.writeNamedWriteable(queryBuilder);
+        }
+    }
```

Comment:
Since it only deals with lists of QueryBuilders I found it more natural to keep it here in the base class for all QueryBuilders rather than blow up StreamInput. I think I saw similar approaches e.g. in the Rounding serialization. But if it is useful elsewhere we can always make it public?